### ISSUE TITLE
Incorrect results when using operator != with tags

### ISSUE THREAD

**Issue by smonkewitz** [influxdb/influxdb#2097]
OS: OSX 10.10.2
Go: version 1.4.2
InfluxDB master commit 0bcec785, built from source

How to reproduce:

```sh
# Create database
curl -G http://localhost:8086/query --data-urlencode "q=CREATE DATABASE bug"

# Write a single point
curl -d '{"database" : "bug", "points": [{"name":"test","tags":{"foo":"bar"},"fields":{"value":1}}]}' -H "Content-Type: application/json" http://localhost:8086/write

# Count records with foo = 'bar'. I get back 1 value, as expected.
curl -G http://localhost:8086/query --data-urlencode "db=bug" --data-urlencode "q=SELECT COUNT(value) FROM test WHERE foo='bar'"
# {"results":[{"series":[{"name":"test","columns":["time","COUNT"],"values":[["1970-01-01T00:00:00Z",1]]}]}]}

# Count records with foo != 'bar'. I still get back 1 value, which is incorrect.
curl -G http://localhost:8086/query --data-urlencode "db=bug" --data-urlencode "q=SELECT COUNT(value) FROM test WHERE foo!='bar'"
# {"results":[{"series":[{"name":"test","columns":["time","COUNT"],"values":[["1970-01-01T00:00:00Z",1]]}]}]}
curl -G http://localhost:8086/query --data-urlencode "db=bug" --data-urlencode "q=SELECT COUNT(value) FROM test WHERE foo<>'bar'"
# {"results":[{"series":[{"name":"test","columns":["time","COUNT"],"values":[["1970-01-01T00:00:00Z",1]]}]}]}
```

**Comment by otoolep** [comment]
Thanks @smonkewitz 

@dgnorton -- should this work?

**Comment by dgnorton** [comment]
@otoolep yes, it should work.  I get the same result shown above.

**Comment by otoolep** [comment]
Working towards fixing this in https://github.com/influxdb/influxdb/pull/2104. We have the same problem with `!=` compares with fields. That has been fixed, now tags need to be fixed.

**Comment by otoolep** [comment]
Fixed by PR #2105

**Comment by otoolep** [comment]
Fixed.
